### ISSUE TITLE
[bb] Specified argument was out of the range of valid values. (Parameter 'index') (Config)

### ISSUE THREAD

**Issue by katameru** [Jackett/Jackett#11002]
### Environment

**OS**:
Linux

**.Net Runtime**: [.Net-Core/.Net-Framework/Mono] whatever is in the `linuxserver/jackett` docker image tagged `sha256:a562e4aafa9860d88dc20f9fbcf1b87a904f4f2ccc6fefc078d2f818f8108104 `
**.Net Version**: whatever is in the `linuxserver/jackett` docker image tagged `sha256:a562e4aafa9860d88dc20f9fbcf1b87a904f4f2ccc6fefc078d2f818f8108104 `
**Jackett Version**: v0.17.449

**Last Working Jackett Version**: N/A

**Are you using a proxy or VPN?** no

### Description

Steps to reproduce: Input login data for bB indexer and click 'Okay'
What happens: Error message appears and indexer is not added.
What should happen: indexer should be configured without error

### Logged Error Messages

```
02-02 18:08:39 Debug WebClient(HttpWebClient2).GetResultAsync(Method: POST Url: https://********.org/login.php PostData: {username=*******, password=*******, keeplogged=1, login=Log In!} RawBody: ) 
02-02 18:08:39 Debug session=DS8zW%2C**************; expires=Sun, 01-Feb-2026 17:08:39 GMT; Max-Age=157680000; path=/; secure; HttpOnly 
02-02 18:08:39 Debug WebClient(HttpWebClient2): Returning Redirect => https://********.org/index.php 0 bytes 
02-02 18:08:39 Debug WebClient(HttpWebClient2).GetResultAsync(Method: GET Url: https://********.org/index.php) 
02-02 18:08:40 Debug session=YgUSM6yvE4wahmc***************; expires=Sun, 01-Feb-2026 17:08:40 GMT; Max-Age=157680000; path=/; secure; HttpOnly 
02-02 18:08:40 Debug WebClient(HttpWebClient2): Returning OK => 8773 bytes
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
    <title>********</title>
    <meta name="ROBOTS" content="NOINDEX,NOFOLLOW" />
    <meta http-equiv="Pragma" content="no-cache" />
    <meta http-equiv="Expires" content="-1" />
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <link rel="shortcut icon" href="favicon.ico" />
    <link href="static/styles/public/style.css" rel="stylesheet" type="text/css" />
    <script src="static/functions/global.js" type="text/javascript"></script>
</head>
<body>
<script type="text/javascript">
      
(function(){
  var global = this;
  var globalName = 'starField';
  var numberOfStars = 30;
  /* total depth of space ;)*/
  var depthDimentsion = 2000;
  /* % of space between browser and viewer.*/
  var viewingDepth = 0.0001;
  /* % of space moved in one step.*/
  var forwardVelocity = 2.5;
  var d = depthDimentsion*(viewingDepth/100);
  var planeDepth = depthDimentsion - d;
  var fv = planeDepth*(forwardVelocity/100);
  var zMultiplier = (depthDimentsion)/d;
  var starObjs, starHTML;
  var posMod, sy, sx, windowCenterY, windowCenterX;
  var scaleXAdjust, scaleYAdjust;
  if((document.layers)&&(this.Layer)){
    starHTML = [
    '<layer id=\"stars','',
    '\" left=\"0\" top=\"0\" width=\"1\" height=\"1"',
    ' bgColor=\"#FFFFFF\"><\/layer>'];
  }else{
    starHTML = [
    '<div id="stars','',
    '" style="position:absolute;width:1px;overflow:',
    'hidden;height:1px;background-color:#FFFFFF;',
    'font-size:1px"><\/div>'];
  }
  function compatModeTest(obj){
    if((document.compatMode)&&
       (document.compatMode.indexOf('CSS') != -1)&&
       (document.documentElement)){
      return document.documentElement;
    }else if(document.body){
      return document.body;
    }else{
      return obj;
    }
  }
  function getWindowState(){
    var global = this;
    var readScroll = {scrollLeft:NaN,scrollTop:NaN};
    var readSizeC = {clientWidth:NaN,clientHeight:NaN};
    var readSizeI = {innerWidth:NaN,innerHeight:NaN};
    var readScrollX = 'scrollLeft';
    var readScrollY = 'scrollTop';
    function getWidthI(){return readSizeI.innerWidth;}
    function getWidthC(){return readSizeC.clientWidth|0;}
    function getHeightI(){return readSizeI.innerHeight;}
    function getHeightC(){return readSizeC.clientHeight|0;}
    function getHeightSmart(){
        return retSmaller(getHeightI(), getHeightC());
    }
    function getWidthSmart(){
        return retSmaller(getWidthI(), getWidthC());
    }
    function setInnerWH(){
      theOne.getWidth = getWidthI;
      theOne.getHeight = getHeightI;
    }
    function retSmaller(inr, other){
      if(other > inr){
        setInnerWH();
        return inr;
      
      }else{
        return other;
      }
    }
    var theOne = {
      getScrollX:function(){return readScroll[readScrollX]|0;},
      getScrollY:function(){return readScroll[readScrollY]|0;},
      getWidth:getWidthC,
      getHeight:getHeightC
    };
    function main(){return theOne;}
    function rankObj(testObj){
      var dv,dhN;
      if(testObj&&(typeof testObj.clientWidth == 'number')&&
         (typeof testObj.clientHeight == 'number')){
        if(((dv = global.innerHeight - testObj.clientHeight) >= 0)&&
           ((dh = global.innerWidth - testObj.clientWidth) >= 0)){
          if(dh == dv){
            return 0;
          }else if((dh&&!dv)||(dv&&!dh)){
            return (dh+dv);
          }
      
}
      }
      return NaN;
    }
    if((typeof global.innerHeight == 'number')&&
       (typeof global.innerWidth == 'number')){
      readSizeI = global;
      var bodyRank = rankObj(document.body);
      var rankDocEl = rankObj(document.documentElement);
      var selEl = null;
      if(!isNaN(bodyRank)){
        if(!isNaN(rankDocEl)){
          if(bodyRank < rankDocEl){
            selEl = document.body;
          }else if(bodyRank > rankDocEl){
            selEl = document.documentElement;
          }else{
            selEl = compatModeTest(document.body);
          }
        }else{
          selEl = document.body;
      
}
      }else if(!isNaN(rankDocEl)){
        selEl = document.documentElement;
      }
      if(selEl){
        readSizeC = selEl
        theOne.getWidth = getWidthSmart;
        theOne.getHeight = getHeightSmart;
      
      }else{
        setInnerWH();
      }
    }else{
      readSizeC = compatModeTest(readSizeC);
    }
    if((typeof global.pageYOffset == 'number')&&
       (typeof global.pageXOffset == 'number')){
      readScroll = global;
      readScrollY = 'pageYOffset';
      readScrollX = 'pageXOffset';
    }else{
      readScroll = compatModeTest(readScroll);
    }
    return (getWindowState = main)();
  }
  var windowState = getWindowState();
  function readWindow(){
    scaleYAdjust = (((windowCenterY =
            (windowState.getHeight() >>1)) - 16)*
                         zMultiplier);
    scaleXAdjust = (((windowCenterX =
            (windowState.getWidth() >> 1)) - 16)*
                        zMultiplier);
    sy = windowCenterY + windowState.getScrollY();
    sx = windowCenterX + windowState.getScrollX();
  }
  function getStyleObj(id){
    var obj = null;
    if(document.getElementById){
      obj = document.getElementById(id);
    }else if(document.all){
      obj = document.all[id];
    }else if(document.layers){
      obj = document.layers[id];
    }
    return ((typeof obj != 'undefined')&&
        (typeof obj.style != 'undefined'))?
                    obj.style:obj;
  }
  function starObj(id, parent, prv){
    var next,reset;
    var divClip, div = getStyleObj("stars"+id);
    var y,x,z,v,dx,dy,dm,dm2,px,py,widthPos,temp;
    (reset = function(){
      px = Math.random()<0.5 ? +1 : -1;
      py = Math.random()<0.5 ? +1 : -1;
      y = ((Math.random()*Math.random()*
          scaleYAdjust)+windowCenterY);
      x = ((Math.random()*Math.random()*
          scaleXAdjust)+windowCenterX);
      widthPos = (x + zMultiplier);
      
      z = 0;
      
;
    z = Math.random()*planeDepth*0.8;
    function step(){
      temp = x * (v = d/(depthDimentsion - z));
      dm = ((dm2 = ((widthPos * v)-temp)|0)>>1);
      dy = (y * v);
      dx = (temp);
    }
      
    if(div){
      if(!posMod){
        posMod = (typeof div.top == 'string')?'px':0;
      }
      divClip =  ((typeof div.clip != 'undefined')&&
               (typeof div.clip != 'string'))?
                       div.clip:div;
      this.position = function(){
        step();
        if(((z += fv) >= planeDepth)||
           ((dy+dm) > windowCenterY)||
          ((dx+dm) > windowCenterX)){
          reset();
          step();
          dm = 0;
      
}
        div.top = ((sy+(py*dy)-dm)|0)+posMod;
        div.left = ((sx+(px*dx)-dm)|0)+posMod;
        divClip.width = (divClip.height = dm2+posMod);
        next.position();
      };
    }else{
      this.position = function(){return;};
    }
    if(++id < numberOfStars){
      next = new starObj(id, parent)
    }else{
      next = parent
    }
  }
  function init(){
    if(!getStyleObj("stars"+(numberOfStars-1))){
      setTimeout(starField, 200);
    }else{
      readWindow();
      starObjs = new starObj(0, init);
      init.act();
    }
  };
  init.position = function(){return;}
  init.act = function(){
    readWindow();
    starObjs.position();
    setTimeout(init.act,50);
  };
  init.act.toString = function(){
    return globalName+'.act()';
  };
  init.toString = function(){
    while(global[globalName])globalName += globalName;
    global[globalName] = this;
    return globalName+'()';
  };
  for(var c = numberOfStars;c--;){
    starHTML[1] = c;
    document.write(starHTML.join(''));
  }
  setTimeout(init, 200);
})();
      
>
<div id="head"></div>
<table cellpadding="0" cellspacing="0" border="0" id="maincontent">
    <tr>
        <td align="center" valign="middle">
            <div id="logo">
<div>
    <object type="application/x-shockwave-flash" data="/static/misc/bacon_invasion.swf" width="540" height="438">
            <param name="movie" value="/static/misc/bacon_invasion.swf" />
    </object>
</div>
                <a href="index.php">Home</a>
                &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="login.php" rel="nofollow">Login</a>
<!--				&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="chat.php" rel="nofollow">Chat</a>-->
            </div>
<div id="subtext">
</div>
        </td>
      
>
    <tr>
        <td align="center">
            Contact: abuse.bb@protonmail.com for enquiries.
        </td>
      
>
</table>
<div id="foot"><div id="copyleft"></div></div>
</body>
</html> 
02-02 18:08:40 Info Executed action Jackett.Server.Controllers.IndexerApiController.UpdateConfig (jackett) in 489.1189ms 
02-02 18:08:40 Info Executed endpoint 'Jackett.Server.Controllers.IndexerApiController.UpdateConfig (jackett)' 
02-02 18:08:40 Error System.ArgumentOutOfRangeException: Specified argument was out of the range of valid values. (Parameter 'index')
   at AngleSharp.Common.ObjectExtensions.GetItemByIndex[T](IEnumerable`1 items, Int32 index)
   at AngleSharp.Dom.HtmlCollection`1.get_Item(Int32 index)
   at Jackett.Common.Indexers.BB.<>c__DisplayClass12_0.<ApplyConfiguration>b__0() in /home/vsts/work/1/s/src/Jackett.Common/Indexers/BB.cs:line 111
   at Jackett.Common.Indexers.BaseIndexer.ConfigureIfOK(String cookies, Boolean isLoggedin, Func`1 onError) in /home/vsts/work/1/s/src/Jackett.Common/Indexers/BaseIndexer.cs:line 240
   at Jackett.Common.Indexers.BB.ApplyConfiguration(JToken configJson) in /home/vsts/work/1/s/src/Jackett.Common/Indexers/BB.cs:line 103
   at Jackett.Server.Controllers.IndexerApiController.UpdateConfig(ConfigItem[] config) in /home/vsts/work/1/s/src/Jackett.Server/Controllers/IndexerApiController.cs:line 97
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfIActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeInnerFilterAsync>g__Awaited|13_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Logged|17_1(ResourceInvoker invoker)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Jackett.Server.Middleware.CustomExceptionHandler.Invoke(HttpContext httpContext) in /home/vsts/work/1/s/src/Jackett.Server/Middleware/CustomExceptionHandler.cs:line 61 System.ArgumentOutOfRangeException: Specified argument was out of the range of valid values. (Parameter 'index')
   at AngleSharp.Common.ObjectExtensions.GetItemByIndex[T](IEnumerable`1 items, Int32 index)
   at AngleSharp.Dom.HtmlCollection`1.get_Item(Int32 index)
   at Jackett.Common.Indexers.BB.<>c__DisplayClass12_0.<ApplyConfiguration>b__0() in /home/vsts/work/1/s/src/Jackett.Common/Indexers/BB.cs:line 111
   at Jackett.Common.Indexers.BaseIndexer.ConfigureIfOK(String cookies, Boolean isLoggedin, Func`1 onError) in /home/vsts/work/1/s/src/Jackett.Common/Indexers/BaseIndexer.cs:line 240
   at Jackett.Common.Indexers.BB.ApplyConfiguration(JToken configJson) in /home/vsts/work/1/s/src/Jackett.Common/Indexers/BB.cs:line 103
   at Jackett.Server.Controllers.IndexerApiController.UpdateConfig(ConfigItem[] config) in /home/vsts/work/1/s/src/Jackett.Server/Controllers/IndexerApiController.cs:line 97
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfIActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeInnerFilterAsync>g__Awaited|13_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Logged|17_1(ResourceInvoker invoker)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Jackett.Server.Middleware.CustomExceptionHandler.Invoke(HttpContext httpContext) in /home/vsts/work/1/s/src/Jackett.Server/Middleware/CustomExceptionHandler.cs:line 61
02-02 18:08:40 Debug The response will be compressed with 'gzip'. 
02-02 18:08:40 Info Request finished HTTP/1.1 POST http://jackett.lan/api/v2.0/indexers/bb/config application/json 211 - 500 - application/json 498.9347ms 
```

**Comment by katameru** [comment]
The username/password I provide is correct for sure because when I input incorrect information I get the appropriate message.
As far as I can tell what happens is:
1. Jackett issues a `POST` to bB
2. It receives a `302 Found` redirection to `/index.php` with a particular session cookie.
3. It issues a `GET` to `index.php` and gets a response with a different session cookie.
4. Since the second session cookie doesn't correspond to a logged-in session Jackett thinks the login failed
5. Since the login didn't fail we're not on `login.php` so the error-building code fails.

So we have two separate bugs here:

- something wonky with the login process, I'm not really sure what's going on there
- failure handling code is too brittle

**Comment by ilike2burnthing** [comment]
Any special characters in your password? If so, try changing it.

**Comment by katameru** [comment]
No special characters

**Comment by ilike2burnthing** [comment]
Can you access the site at the moment? I'm getting `Database error`. I don't have an account, so I'm not sure if that is a result of the site using an IP whitelist or something else.

**Comment by katameru** [comment]
Yes, the site is accessible and I can log in through the browser without issue. The site does not employ any kind of whitelisting.

**Comment by ilike2burnthing** [comment]
@janot / @mueslo any issues?

**Comment by janot** [comment]
No problems on Windows

**Comment by mueslo** [comment]
> @janot / @mueslo any issues?

Can't reproduce (0.16.1902.0)

**Comment by katameru** [comment]
It seems that something is indeed broken with the tracker login flow and not with Jackett. The issue only shows up in very specific circumstances in the browser but pretty often with tools.

The workaround is to keep retrying the login, it has probably around a 10% chance of working. Once you're logged in it should be fine.

I'll see if this will be fixed on the tracker end. If not, maybe some retry logic in the indexer might help.

**Comment by ilike2burnthing** [comment]
Thanks all.

@mueslo are you purposefully using an old version (e.g. waiting for a fix)? If not, current is [v0.17.449](https://github.com/Jackett/Jackett/releases/tag/v0.17.449).

@katameru Let us know if you can't trace the issue to elsewhere. The label applied above will close the issue in 7 days if you don't reply before then (though you can always reopen if needed).

**Comment by katameru** [comment]
The issue has been fixed on the tracker side.
